frontend/src/app/core/services/firebase-auth.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable, NgZone } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import { GithubAuthProvider, GoogleAuthProvider } from 'firebase/auth';
+import firebase from 'firebase/compat';
+import { first, from, Observable, switchMap, tap, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class FirebaseAuthService {
+    private tokenKeyName = 'user-token';
+
+    constructor(private afAuth: AngularFireAuth, private ngZone: NgZone) {
+        afAuth.authState.subscribe(async (user) => {
+            if (user) {
+                localStorage.setItem(this.tokenKeyName, await user.getIdToken());
+            } else {
+                localStorage.removeItem(this.tokenKeyName);
+            }
```

Comment:
```suggestion
            user
                ? localStorage.setItem(this.tokenKeyName, await user.getIdToken())
                :  localStorage.removeItem(this.tokenKeyName);
```

Reply:
### ESLint doesn't allow it

![Screenshot 2023-08-22 165351](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/106184656/b62fec4a-6ede-472f-b2ef-42d6b74ab0dd)
![Screenshot 2023-08-22 165431](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/106184656/9ff1caff-038b-4652-8afc-c45dbbdfab63)